types/eslint/tsconfig.json

```diff
@@ -11,7 +11,8 @@
         "strictFunctionTypes": true,
         "types": [],
         "noEmit": true,
-        "forceConsistentCasingInFileNames": true
+        "forceConsistentCasingInFileNames": true,
+        "exactOptionalPropertyTypes": true
```

Comment:
[Praise] Yes, great!

Reply:
Is this required for something? Usually it does not matter to set this on DT because we don't have all that much "runtime code".

Reply:
It's only for testing. In some cases ESLint won't care if a property is not set, but it will raise an error if that property is explicitly set to `undefined`. I tried to model this behavior in the types and wrote some test that would fail if a property is `undefined` [here](https://github.com/fasttime/DefinitelyTyped/blob/efcea5d15da4f77099d1cf6df2cca53e306ad95c/types/eslint/eslint-tests.ts#L1279-L1282) and [here](https://github.com/fasttime/DefinitelyTyped/blob/efcea5d15da4f77099d1cf6df2cca53e306ad95c/types/eslint/eslint-tests.ts#L1414-L1419), and it's those tests that require `exactOptionalPropertyTypes`. But I don't think there's a way to enforce the compiler errors will also occur on consumer side if users don't have `"exactOptionalPropertyTypes": true` in their config.